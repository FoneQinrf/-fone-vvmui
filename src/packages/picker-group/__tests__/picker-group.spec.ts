/*
 * @Author: Fone`峰
 * @Date: 2021-06-01 12:55:23
 * @LastEditors: Fone`峰
 * @LastEditTime: 2021-06-03 11:41:35
 * @Description: file content
 * @Email: qinrifeng@163.com
 * @Github: https://github.com/FoneQinrf
 */
import { mount } from '@vue/test-utils';
import PickerGroup from '../';
import { nextTick } from 'vue';

const columns = [
  {
    value: 'jiangsu',
    label: '浙江',
    children: [
      {
        value: 'nanjing',
        label: '杭州',
        children: [
          {
            value: 'fuzimiao',
            label: '西湖'
          }
        ]
      },
      {
        value: 'suzhou',
        label: '宁波',
        children: [
          {
            value: 'zhuozhengyuan',
            label: '老庙'
          },
          {
            value: 'shizilin',
            label: '老街'
          }
        ]
      }
    ]
  },
  {
    value: 'jiangsu',
    label: '江苏',
    children: [
      {
        value: 'nanjing',
        label: '南京',
        children: [
          {
            value: 'fuzimiao',
            label: '夫子庙'
          }
        ]
      },
      {
        value: 'suzhou',
        label: '苏州',
        children: [
          {
            value: 'zhuozhengyuan',
            label: '拙政园'
          },
          {
            value: 'shizilin',
            label: '狮子林'
          }
        ]
      }
    ]
  }
];

describe('PickerGroup', () => {
  it('should columns render corretly', async () => {
    const wrapper = mount(PickerGroup, {
      props: { columns: columns, teleport: '' }
    });
    wrapper.trigger('click');
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findAll('.vvm-picker').length === 3).toBe(true);
    expect(wrapper.vm.count).toEqual([0, 0, 0]);
  });

  it('should v-model and chooseText render corretly', async () => {
    const wrapper = mount(PickerGroup, {
      props: { columns: columns, teleport: '', modelValue: [1, 1, 1] }
    });
    wrapper.trigger('click');
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.vm.count).toEqual([1, 1, 1]);
    expect(wrapper.vm.chooseText === '江苏 / 苏州 / 狮子林').toBe(true);
  });

  it('should confirm and cancel click corretly', async () => {
    const wrapper = mount(PickerGroup, {
      props: { columns: columns, teleport: '', modelValue: [1, 1, 1] }
    });
    wrapper.trigger('click');
    await nextTick();
    wrapper.find('.vvm-picker-group__btu.__cancel').trigger('click');
    wrapper.find('.vvm-picker-group__btu.__confirm').trigger('click');
    expect(wrapper.emitted()).toHaveProperty('confirm');
    expect(wrapper.emitted()).toHaveProperty('cancel');
  });

  it('should close render corretly', async () => {
    const wrapper = mount(PickerGroup, {
      props: { columns: columns, teleport: '', modelValue: [1, 1, 1] }
    });
    wrapper.trigger('click');
    await nextTick();
    wrapper.find('.vvm-picker-group__btu.__confirm').trigger('click');
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('.vvm-picker-group__body').exists()).toBe(false);
  });

  it('should async columns and modelValue render corretly', async () => {
    const wrapper = mount(PickerGroup, {
      props: { teleport: '' }
    });
    wrapper.trigger('click');
    await nextTick();
    wrapper.setProps({ columns });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findAll('.vvm-picker').length === 3).toBe(true);
    expect(wrapper.vm.count).toEqual([0, 0, 0]);
    wrapper.setProps({ modelValue: [1, 1, 1] });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.vm.count).toEqual([1, 1, 1]);
    expect(wrapper.vm.chooseText === '江苏 / 苏州 / 狮子林').toBe(true);
  });
});
