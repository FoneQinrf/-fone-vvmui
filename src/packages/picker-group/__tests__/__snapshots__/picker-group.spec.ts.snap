// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerGroup should async columns and modelValue render corretly 1`] = `
"<div class=\\"vvm-picker-group\\">
  <div class=\\"vvm-input vvm-input--right\\">
    <!--v-if--><input class=\\"vvm-input--text\\" type=\\"text\\" readonly=\\"\\" placeholder=\\"请选择\\">
    <!--v-if-->
  </div>
  <!--teleport start-->
  <transition-stub>
    <div class=\\"vvm-popup--overlay\\"></div>
  </transition-stub>
  <transition-stub>
    <div class=\\"vvm-popup vvm-popup--bottom\\">
      <div class=\\"vvm-picker-group__body\\">
        <div class=\\"vvm-picker-group__top\\"><span class=\\"vvm-picker-group__btu __cancel\\">取消</span><span class=\\"vvm-picker-group__btu __confirm\\">确认</span></div>
        <div class=\\"vvm-picker-group__columns\\">
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(108px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">浙江</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">江苏</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(108px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">杭州</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">宁波</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(108px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">西湖</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <!--teleport end-->
</div>"
`;

exports[`PickerGroup should async columns and modelValue render corretly 2`] = `
"<div class=\\"vvm-picker-group\\">
  <div class=\\"vvm-input vvm-input--right\\">
    <!--v-if--><input class=\\"vvm-input--text\\" type=\\"text\\" readonly=\\"\\" placeholder=\\"请选择\\">
    <!--v-if-->
  </div>
  <!--teleport start-->
  <transition-stub>
    <div class=\\"vvm-popup--overlay\\"></div>
  </transition-stub>
  <transition-stub>
    <div class=\\"vvm-popup vvm-popup--bottom\\">
      <div class=\\"vvm-picker-group__body\\">
        <div class=\\"vvm-picker-group__top\\"><span class=\\"vvm-picker-group__btu __cancel\\">取消</span><span class=\\"vvm-picker-group__btu __confirm\\">确认</span></div>
        <div class=\\"vvm-picker-group__columns\\">
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(72px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">浙江</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">江苏</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(72px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">南京</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">苏州</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(72px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">拙政园</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">狮子林</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <!--teleport end-->
</div>"
`;

exports[`PickerGroup should close render corretly 1`] = `
"<div class=\\"vvm-picker-group\\">
  <div class=\\"vvm-input vvm-input--right\\">
    <!--v-if--><input class=\\"vvm-input--text\\" type=\\"text\\" readonly=\\"\\" placeholder=\\"请选择\\">
    <!--v-if-->
  </div>
  <!--teleport start-->
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <!--teleport end-->
</div>"
`;

exports[`PickerGroup should columns render corretly 1`] = `
"<div class=\\"vvm-picker-group\\">
  <div class=\\"vvm-input vvm-input--right\\">
    <!--v-if--><input class=\\"vvm-input--text\\" type=\\"text\\" readonly=\\"\\" placeholder=\\"请选择\\">
    <!--v-if-->
  </div>
  <!--teleport start-->
  <transition-stub>
    <div class=\\"vvm-popup--overlay\\"></div>
  </transition-stub>
  <transition-stub>
    <div class=\\"vvm-popup vvm-popup--bottom\\">
      <div class=\\"vvm-picker-group__body\\">
        <div class=\\"vvm-picker-group__top\\"><span class=\\"vvm-picker-group__btu __cancel\\">取消</span><span class=\\"vvm-picker-group__btu __confirm\\">确认</span></div>
        <div class=\\"vvm-picker-group__columns\\">
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(108px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">浙江</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">江苏</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(108px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">杭州</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">宁波</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(108px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">西湖</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <!--teleport end-->
</div>"
`;

exports[`PickerGroup should v-model and chooseText render corretly 1`] = `
"<div class=\\"vvm-picker-group\\">
  <div class=\\"vvm-input vvm-input--right\\">
    <!--v-if--><input class=\\"vvm-input--text\\" type=\\"text\\" readonly=\\"\\" placeholder=\\"请选择\\">
    <!--v-if-->
  </div>
  <!--teleport start-->
  <transition-stub>
    <div class=\\"vvm-popup--overlay\\"></div>
  </transition-stub>
  <transition-stub>
    <div class=\\"vvm-popup vvm-popup--bottom\\">
      <div class=\\"vvm-picker-group__body\\">
        <div class=\\"vvm-picker-group__top\\"><span class=\\"vvm-picker-group__btu __cancel\\">取消</span><span class=\\"vvm-picker-group__btu __confirm\\">确认</span></div>
        <div class=\\"vvm-picker-group__columns\\">
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(72px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">浙江</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">江苏</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(72px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">南京</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">苏州</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
          <div class=\\"vvm-picker\\" style=\\"height: 252px;\\">
            <div class=\\"vvm-picker__wrp\\">
              <div class=\\"vvm-picker__scroll\\" style=\\"height: 252px; transform: translateY(72px);\\">
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">拙政园</p>
                <p style=\\"height: 36px;\\" class=\\"vvm-picker__item vvm-ellipsis\\">狮子林</p>
              </div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --top vvm-hairline-bottom\\"></div>
              <div style=\\"height: 108px;\\" class=\\"vvm-picker__mask --bottom vvm-hairline-top\\"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <!--teleport end-->
</div>"
`;
